nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -858,11 +874,15 @@ ngx_http_modsecurity_handler(ngx_http_request_t *r)
     ngx_http_modsecurity_loc_conf_t *cf;
     ngx_http_modsecurity_ctx_t      *ctx;
     ngx_int_t                        rc;
+    ngx_time_t                      *start_time;
+
```

Comment:
Shouldnt we do this ngx_time_update() here?

Reply:
ngx_timeofday() gives the cached time and is not updated.  It gives access to cached time value.
So in order to get exact time within the function, I am manually updating the time.

Reply:
I'm a bit confused with the conversation here. I think @venkatsvpr is suggesting that we should call `ngx_time_update()` before calling `ngx_timeofday()`, for the exact same reasoning that you are just describing?

Reply:
updated the same. Calling it before also now